zellij-server/src/screen.rs

```diff
@@ -1257,232 +1088,139 @@ pub(crate) fn screen_thread_main(
-                }
-
-                screen.render();
+                active_tab!(screen, client_id, |tab: &mut Tab| tab
+                    .copy_selection(client_id));
             }
             ScreenInstruction::Exit => {
-                break;
+                return;
             }
             ScreenInstruction::ToggleTab(client_id) => {
                 screen.toggle_tab(client_id);
-                screen
-                    .bus
-                    .senders
-                    .send_to_server(ServerInstruction::UnblockInputThread)
-                    .unwrap();
-
-                screen.render();
+                screen.unblock_input();
             }
             ScreenInstruction::AddClient(client_id) => {
                 screen.add_client(client_id);
                 screen.update_tabs();
-
-                screen.render();
+                continue;
             }
             ScreenInstruction::RemoveClient(client_id) => {
                 screen.remove_client(client_id);
-
-                screen.render();
             }
             ScreenInstruction::AddOverlay(overlay, _client_id) => {
                 screen.get_active_overlays_mut().pop();
                 screen.get_active_overlays_mut().push(overlay);
-                screen
-                    .bus
-                    .senders
-                    .send_to_server(ServerInstruction::UnblockInputThread)
-                    .unwrap();
+                screen.unblock_input();
+                continue;
             }
             ScreenInstruction::RemoveOverlay(_client_id) => {
                 screen.get_active_overlays_mut().pop();
-                screen.render();
-                screen
-                    .bus
-                    .senders
-                    .send_to_server(ServerInstruction::UnblockInputThread)
-                    .unwrap();
+                screen.unblock_input();
             }
             ScreenInstruction::ConfirmPrompt(_client_id) => {
                 let overlay = screen.get_active_overlays_mut().pop();
                 let instruction = overlay.and_then(|o| o.prompt_confirm());
                 if let Some(instruction) = instruction {
                     screen.bus.senders.send_to_server(*instruction).unwrap();
                 }
-                screen
-                    .bus
-                    .senders
-                    .send_to_server(ServerInstruction::UnblockInputThread)
-                    .unwrap();
+                screen.unblock_input();
+                continue;
             }
             ScreenInstruction::DenyPrompt(_client_id) => {
                 screen.get_active_overlays_mut().pop();
-                screen.render();
-                screen
-                    .bus
-                    .senders
-                    .send_to_server(ServerInstruction::UnblockInputThread)
-                    .unwrap();
+                screen.unblock_input();
             }
         }
+        screen.render();
```

Comment:
I think this refactoring is pretty good.

But since `screen.render()` comes to the bottom, some Instruction seems to need to `continue` and you already know that.
It would be nice to add `continue` missing in some Instruction. (`RemoveOverlay`, `DenyPrompt`)

Reply:
Thanks for the feedback.

> It would be nice to add `continue` missing in some Instruction. (`RemoveOverlay`, `DenyPrompt`)

Looking at the diff, `screen.render()` was previously called in these functions, too. It's just above the `screen.bus.senders ...` block which makes it sort of easy to miss. If the `screen.render()` there was wrong before I can of course add `continue` statements, but I didn't want to "correct" code that I don't fully understand. :)

Reply:
Oh! sorry for confusing. I think I missed it. Those instructions have to call `screen.render()`. Thanks 👍